README.md

```diff
@@ -194,6 +194,11 @@ normalization to JSON API.
   - `options.params` defaults to `null`. The parameters to include, either in the URL (for `GET`
     requests) or request body (for others).
 
+  - `options.urlParams` defaults to `null`. The parameters that will be
```

Comment:
Why `urlParams` here vs something like `queryParams`? Using `urlParams` seems somewhat odd to me because (at least for `GET` requests) `options.params` is _also_ put in the URL.

Reply:
Kinda agree with you, the `urlParams` indeed does the same thing as `params` for `GET` requests. The initial purpose of #782 is to split the `params` to `urlParams` which will be appended to the URL as the query string and body `payload`. The current solution will cover this purpose, but it is sort of redundant, I think that @hjdivad suggests this way mainly for the backward compatiblity.

I've thought about some alternatives, e.g.
- Remove the `params` option, add `urlParams` and `payload`.
- For `GET` requests, it could only has one of `urlParams` or `params`, or an error will be thrown. Or even further, it could only use `urlParams` for `GET` requests.

Reply:
@2hu12 I think @rwjblue was mainly suggesting a different name rather than a different implementation.  I agree that `queryParams` is a more typical name than `urlParams`.  I should have thought a bit more before making that suggestion, sorry.

You're right that I was thinking about backwards compatibility.  But maybe the better approach is to do your original suggestion of `queryParams` and `payload` / `body` and keep `options.params` but deprecated.  What do you think?

I really don't want to impose restrictions based on the request method because not having a request body for `GET` requests is just a convention that servers need not honour.

Reply:
@hjdivad Thanks for the explanation.

I think my original purpose( add `queryParams` and `payload` options and deprecate `params` option) won't be implemented only in `ember-m3`, for that the `params` is actually used by the `RESTAdapter.ajax` and it will be treated differently for different request methods, which is also related to the `GET` payload restrictions, for example, even if we want to provide the `payload` option for one `GET` request, it is impossible currently.

I just update the `urlParams` to `queryParams` for now, and maybe if one day `RESTAdapter.ajax` add options for `queryParams` and `payload`, we could follow it.